General cleanup with eslint updates

brodybits opened the pull request:
UPDATED after review and discussion:

- ~~merge changes from PR #81 to remove standard eslint exceptions from `.eslint.yml` that are not needed~~
- manual cleanup fix for `no-prototype-builtins` (as needed by new (semi)standard eslint package updates)
- eslint-config-semistandard@14 & eslint-config-standard@13 updates with automatic cleanup fixes (since the `prefer-const` & `quote-props` eslint rules are now enabled)
- automatic eslint cleanup fixes for `padded-blocks` that is no longer kept as an exception

I tried to keep these changes split into separate commits in a semi-sensical manner. I think it would be _a little_ too much to put all these changes into a single commit. I am _now_ proposing _the most important_ changes in _this_ PR for discussion, _with some apparently controversial changes now split into separate PRs_.

My objective is to keep the number of (semi)standard eslint exceptions to a bare minimum.

raphinesse commented:
@janpio I think consistency with other repos should not be a matter here, if we see this as a test drive towards a better, stricter code style.

brodybits commented:
I just pushed the more important changes in a different order, in a cleaner sequence, and rebased on `master`. I moved the seemingly controversial proposals out to separate PRs #85 & #86. I would really appreciate another review.

I pushed the original proposal into my [`general-cleanup-with-eslint-update-save1`](https://github.com/brodybits/cordova-common/tree/general-cleanup-with-eslint-update-save1) branch in case we want to keep it for comparison..

raphinesse commented on src/CordovaError/CordovaError.js:
```diff
@@ -50,10 +50,8 @@ CordovaError.EXTERNAL_TOOL_ERROR = 1;
  */
 CordovaError.prototype.getErrorCodeName = function () {
     for (var key in CordovaError) {
```
```suggestion
     for (const key of Object.keys(CordovaError)) {
```

I noticed this before but forgot to add it to my review. I'm so sorry about that.

With the change proposed above, we should be preserving the exact behavior of the changed code. That is, iterating only over own keys.

brodybits commented:
I applied the change suggested by @raphinesse, with the code alignment fixed, preserved these updates in my [`general-cleanup-with-eslint-update-save2`](https://github.com/brodybits/cordova-common/tree/general-cleanup-with-eslint-update-save2) branch, then rebased the changes as 3 commits on top of the `master` branch.

I think these changes should be good to go, would appreciate a final review.

I am not sure whether these changes should be merged by a merge commit or rebase commit (I would not favor a squash merge). The merge commit would clearly show that these changes go together with this PR number, while the rebase commit might be a little less noisy. (GitHub would still show the PR number for each of the commits, though.)
